examples/tool_tests/gowin.c

```diff
@@ -1,10 +1,13 @@
-#pragma PART "GW2AR-18C GW2AR-LV18QN88PC8/I7"
+// @tiagosr: adding --device_version (or --device_name) at the end as 
+// PipelineC currently recognizes vendors by the first letters of the part name
+#pragma PART "GW2AR-LV18QN88PC8/I7 --device_version C"
 #pragma MAIN_MHZ my_pipeline 100.0
 // TEMP NOT UNTIL DONE DEV #include "examples/pipeline.c"
 #include "intN_t.h"
 #include "uintN_t.h"
 #pragma MAIN my_pipeline
-uint32_t my_pipeline(uint32_t x, uint32_t y)
+// @tiagosr: sized ports down to fit the amount of I/O in the device
```

Comment:
Yes, this is somewhat critical for being able to synthesize larger pipelines.

Vivado has a mode called 'out of context' which does not assign top level IO to actual device pins. Quartus calls this 'virtual pins'. nextpnr only supports this for lattice ecp5 also called `--out-of-context`

That said, I have not seen how to do this in Efinix Efinity software for example. So its possible GOWIN tools don't support it? - but maybe they do? - need to read docs etc ...

Reply:
I'll try finding something to that effect - a preliminary search in the docs and in gw_sh's inline help didn't bring much useful info on that yet :/

Reply:
:fist_raised: appreciate you looking

Reply:
So, the `run syn` command doesn't complain with larger amounts of IO, and outputs the timing estimates in HTML that I mentioned in another thread. I'll work that into the parser instead, then.

Reply:
:hushed: ooo thats very neat to hear - appreciate you modifying to use that, sounds like it will work great

Reply:
I'll remove that comment soon - just completed the code to parse the html report 😄